cypress/e2e/dashboard/chapters/chapter-administrator.cy.ts

```diff
@@ -0,0 +1,237 @@
+        ({ id, instance_role }) => {
+          expect(instance_role.name).to.eq(instanceRoles.MEMBER);
+          sequenceChangeChapterUserRole(id, [
+            { chapterId: firstChapterId, roleName: chapterRoles.ADMINISTRATOR },
+          ]);
+          cy.changeInstanceUserRole({
+            roleName: instanceRoles.OWNER,
+            userId: id,
+          }).then(expectNoErrors);
+
+          confirmInstanceRole(users.testUser.email, instanceRoles.OWNER);
+          cy.changeInstanceUserRole({
+            roleName: instanceRoles.MEMBER,
+            userId: id,
+          }).then(expectNoErrors);
+        },
+      );
+      confirmInstanceRole(
+        users.testUser.email,
+        instanceRoles.CHAPTER_ADMINISTRATOR,
+      );
+    });
+
+    it('should be kept when changing chapter_administrator instance role to member for administrator of chapter', () => {
+      cy.login();
+      cy.task<User>('getUser', users.testUser.email).then(
+        ({ id, instance_role }) => {
+          expect(instance_role.name).to.eq(instanceRoles.MEMBER);
+          sequenceChangeChapterUserRole(id, [
+            { chapterId: firstChapterId, roleName: chapterRoles.ADMINISTRATOR },
+          ]);
+          confirmInstanceRole(
+            users.testUser.email,
+            instanceRoles.CHAPTER_ADMINISTRATOR,
+          );
+
+          cy.changeInstanceUserRole({
+            roleName: instanceRoles.MEMBER,
+            userId: id,
+          }).then(expectNoErrors);
+        },
+      );
+      confirmInstanceRole(
+        users.testUser.email,
+        instanceRoles.CHAPTER_ADMINISTRATOR,
+      );
+    });
+
+    it('should be changed to owner when changing instance role to owner, for administrator of chapter', () => {
+      cy.login();
+      cy.task<User>('getUser', users.testUser.email).then(
+        ({ id, instance_role }) => {
+          expect(instance_role.name).to.eq(instanceRoles.MEMBER);
+          sequenceChangeChapterUserRole(id, [
+            { chapterId: firstChapterId, roleName: chapterRoles.ADMINISTRATOR },
+          ]);
+          confirmInstanceRole(
+            users.testUser.email,
+            instanceRoles.CHAPTER_ADMINISTRATOR,
+          );
+
+          cy.changeInstanceUserRole({
+            roleName: instanceRoles.OWNER,
+            userId: id,
+          }).then(expectNoErrors);
+        },
+      );
+      confirmInstanceRole(users.testUser.email, instanceRoles.OWNER);
+    });
+
+    it('should not be visible on users list', () => {
+      cy.login();
+      cy.visit('/dashboard/users');
+      cy.findByRole('table', { name: 'Instance Users' }).should('be.visible');
+      cy.contains('chapter_administrator').should('not.exist');
+    });
+  });
+
+  function confirmInstanceRole(email: string, expectedRole: string) {
+    cy.task<User>('getUser', email).then(({ instance_role: { name } }) => {
+      expect(name).to.eq(expectedRole);
+    });
+  }
+
+  function sequenceChangeChapterUserRole(
```

Comment:
How about just `applyChapterRoleChanges`?   I wasn't sure what 'sequence' meant in this content and it looks like it's just 'do this' :)

Reply:
Thanks for the idea, I wasn't happy with it and your version is better.